tools/check_spelling.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# Applies requisite code formatters to the source tree
+# check_spelling.sh
+
+set -e
+
+# Used to filter a list of words that do not require spell checking
+IGNORE_STRING="Creater,creater,ect,overriden"
+
+MISSPELL_ARGS="-error -o stderr"
+
+if [[ $1 == "fix" ]];then
+  MISSPELL_ARGS="-w"
+fi
+
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+ROOTDIR=$SCRIPTPATH/..
+cd "$ROOTDIR"
+
+GOPATH=$(cd "/tmp/"; pwd)
+export GOPATH
+export PATH=$GOPATH/bin:$PATH
+
+# Install tools we need
+go get -u github.com/client9/misspell/cmd/misspell
```

Comment:
Can this be part of CI image building process and downloading a released version? Getting master every time will cause inconsistent behavior.